File: web-portal-react-client/src/services/ClientExpenseService.js
import BaseService from './BaseService'

import {
    PAGINATION,
    ALLOWED_FILE_FORMATS,
    ALLOWED_FILE_FORMAT_MIME_TYPES
} from 'lib/Constants'

const { FIRST_PAGE } = PAGINATION
const { ZIP } = ALLOWED_FILE_FORMATS

export class ClientExpenseService extends BaseService {
    find({ clientId, page = FIRST_PAGE, size = 10, ...other }) {
        return super.request({
            url: `/clients/${clientId}/expenses`,
            params: { page: page - 1, size, ...other }
        })
    }

    findById(expenseId, { clientId }) {
        return super.request({
            url: `/clients/${clientId}/expenses/${expenseId}`,
            response: { extractDataOnly: true }
        })
    }

    count({ clientId, ...params }) {
        return super.request({
            url: `/clients/${clientId}/expenses/count`,
            response: { extractDataOnly: true },
            params
        })
    }

    total({ clientId, ...params }) {
        return super.request({
            url: `/clients/${clientId}/expenses/total`,
            response: { extractDataOnly: true },
            params
        })
    }

    canView({ clientId, ...params }) {
        return super.request({
            url: `/clients/${clientId}/expenses/can-view`,
            response: { extractDataOnly: true },
            params
        })
    }

    canAdd({ clientId, ...params }) {
        return super.request({
            url: `/clients/${clientId}/expenses/can-add`,
            response: { extractDataOnly: true },
            params
        })
    }

    save(data, { clientId }) {
        return super.request({
            method: data.id ? 'PUT' : 'POST',
            url: `/clients/${clientId}/expenses`,
            body: data,
            type: 'multipart/form-data',
            responseTimeout: 1200000,
            response: { extractDataOnly: true }
        })
    }
}

const service = new ClientExpenseService()
export default service

File: EldermarkExchangeNew/eldermark-exchange-core-services/src/main/java/com/scnsoft/eldermark/service/client/expense/ClientExpenseService.java
package com.scnsoft.eldermark.service.client.expense;

import com.scnsoft.eldermark.entity.client.expense.ClientExpense;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface ClientExpenseService {

    ClientExpense save(ClientExpense entity);

    Optional<ClientExpense> findById(Long expenseId);

    <P> Optional<P> findById(Long expenseId, Class<P> projectionClass);

    <P> Page<P> find(Long clientId, Pageable pageable, Class<P> projectionClass);

    Long count(Long clientId);

    Long getTotalCost(Long clientId);
}
